rpc_util.go

```diff
@@ -817,16 +817,43 @@ func decompress(compressor encoding.Compressor, d []byte, maxReceiveMessageSize
 			//
 			// TODO: If we ensure that the buffer size is the same as the DecompressedSize,
 			// we can also utilize the recv buffer pool here.
-			buf := bytes.NewBuffer(make([]byte, 0, size+bytes.MinRead))
-			bytesRead, err := buf.ReadFrom(io.LimitReader(dcReader, int64(maxReceiveMessageSize)+1))
+			bufferSize := uint64(size) + bytes.MinRead
+			if bufferSize > math.MaxInt {
+				bufferSize = math.MaxInt
+			}
+			buf := bytes.NewBuffer(make([]byte, 0, int(bufferSize)))
+			bytesRead, err := buf.ReadFrom(io.LimitReader(dcReader, int64(maxReceiveMessageSize)))
+			if err != nil {
+				return nil, int(bytesRead), err
+			}
+
+			if err = checkReceiveMessageOverflow(bytesRead, int64(maxReceiveMessageSize), dcReader); err != nil {
+				return nil, size + 1, err
+			}
 			return buf.Bytes(), int(bytesRead), err
+
 		}
 	}
 	// Read from LimitReader with limit max+1. So if the underlying
 	// reader is over limit, the result will be bigger than max.
-	d, err = io.ReadAll(io.LimitReader(dcReader, int64(maxReceiveMessageSize)+1))
+	d, err = io.ReadAll(io.LimitReader(dcReader, int64(maxReceiveMessageSize)))
```

Comment:
can you change the above comment relevant to your change? Since, we are not reading max+1 anymore.

Explain that we are checking the overflow separately after

Reply:
done

Reply:
@infovivek2020 you don't have to write docstring in inline comments. The only thing you need to update is remove the existing comment and write comment over `checkReceiveMessageOverflow` which is to check overflow.

Also, please don't resolve comment by yourself. We follow the practice of reviewer resolving comments after the change